### ISSUE TITLE
mutating_webhook_configuration and validation_webhook_configuration error with no rules

### ISSUE THREAD

**Issue by onwsk8r** [hashicorp/terraform-provider-kubernetes#1643]
### Terraform Version, Provider Version and Kubernetes Version

```
❯ terraform -v
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/kubernetes v2.8.0
Kubernetes Version: 1.21.6
```

### Affected Resource(s)
- kubernetes_mutating_webhook_configuration
- kubernetes_validating_webhook_configuration

### Terraform Configuration Files
```hcl
resource "kubernetes_mutating_webhook_configuration" "example" {
  metadata {
    name = "foo"
  }

  webhook {
    name = "foo"

    client_config {
      service {
        namespace = "default"
        name      = "foo-webhook"
      }
    }

    failure_policy            = "Fail"
    side_effects              = "None"
    timeout_seconds           = 2
    admission_review_versions = ["v1beta1"]
  }
}

resource "kubernetes_validating_webhook_configuration" "example" {
  metadata {
    name = "foo"
  }

  webhook {
    name = "foo"

    client_config {
      service {
        namespace = "default"
        name      = "foo-webhook"
      }
    }

    failure_policy            = "Fail"
    side_effects              = "None"
    timeout_seconds           = 2
    admission_review_versions = ["v1beta1"]
  }
}
```

### Debug Output
tl;dr
```
2022-03-15T05:48:58.906-0500 [TRACE] vertex "kubernetes_validating_webhook_configuration.example": starting visit (*terraform.NodeValidatableResource)
2022-03-15T05:48:58.906-0500 [TRACE] vertex "kubernetes_mutating_webhook_configuration.example": starting visit (*terraform.NodeValidatableResource)
2022-03-15T05:48:58.906-0500 [ERROR] vertex "kubernetes_validating_webhook_configuration.example" error: Insufficient rule blocks
2022-03-15T05:48:58.906-0500 [TRACE] vertex "kubernetes_validating_webhook_configuration.example": visit complete, with errors
2022-03-15T05:48:58.906-0500 [ERROR] vertex "kubernetes_mutating_webhook_configuration.example" error: Insufficient rule blocks
2022-03-15T05:48:58.906-0500 [TRACE] vertex "kubernetes_mutating_webhook_configuration.example": visit complete, with errors
```
See the whole thing [here](https://gist.github.com/onwsk8r/c55723b1523ada964c1fd2d212cc4555).

### Steps to Reproduce

1. Put the sample HCl code above into a file
2. Run `terraform init`
3. Run `terraform apply`
4. $Profit (or not)

### Expected Behavior

The Terraform provider is expected to demonstrate similar behavior to the Kubernetes API. HCl code and `terraform apply` should produce the same result as YAML and `kubectl apply`.

### Actual Behavior

Terraform indicates that `At least 1 "rule" blocks are required` (is that proper Engrish?), which is not the case when running `kubectl apply` with the YAML version of the code. Moreover, the documentation lists `rule` blocks as optional.

### Important Factoids

The [resource documentation](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/mutating_webhook_configuration_v1) shows `rule` as being `optional`, which is how the Kubernetes API works as well. **But wait**! The [Dynamic Admission Controllers documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#webhook-configuration) says that `[e]ach webhook must specify a list of rules`! Case closed, you say - the documentation supports the requirement. Indeed it does not, because an empty list is a valid list, and the API docs for [each](https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/mutating-webhook-configuration-v1/) [resource](https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/validating-webhook-configuration-v1/) do not specify a minimum length for the list!

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by arybolovlev** [comment]
Hi @onwsk8r,

Thank you for reporting this issue. It seems to be a duplicate of this [one](https://github.com/hashicorp/terraform-provider-kubernetes/issues/1618). There is [PR](https://github.com/hashicorp/terraform-provider-kubernetes/pull/1628) that addresses this issue. Please keep an eye on our release notes.

Thank you!
